button-examples/Android/android-button-example.md

```diff
@@ -23,213 +21,387 @@ There are four types of buttons:
 
 </details>
 
+<details>
+<summary>Icon</summary>
+<br>
 
+|  | Attribute | Related method(s) | Default value |
+| --- | --- | --- | --- |
+| **Icon** | `app:icon` | `setIcon`<br/>`setIconResource`<br/>`getIcon` | `null` |
+| **Color** | `app:iconTint` | `setIconTint`<br/>`setIconTintResource`<br/>`getIconTint` | `?attr/colorPrimary` |
+| **Size** | `app:iconSize` | `setIconSize`<br/>`getIconSize` | `wrap_content` |
+| **Gravity** (position relative to text label) | `app:iconGravity` | `setIconGravity`<br/>`getIconGravity` | `start` |
+| **Padding** (space between icon and text label) | `app:iconPadding` | `setIconPadding`<br/>`getIconPadding` | `4dp` |
+
+</details>
+
+See the full list of [attrs](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/button/res/values/attrs.xml).
+
+</details>
+
+<details>
+<summary><b>Styles</b></summary>
+<br>
+
+|  | Style |
+| --- | --- |
+| **Default style** | `Widget.MaterialComponents.Button.TextButton` |
+| **Icon style** | `Widget.MaterialComponents.Button.TextButton.Icon` |
+
+See the full list of [styles](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/button/res/values/styles.xml).
+
+Default style theme attribute: N/A
+
+</details>
 
 ## Outlined button
 
 [Outlined buttons](https://material.io/components/buttons/#outlined-button) are medium-emphasis buttons. They contain actions that are important, but aren’t the primary action in an app.
 
+### Outlined button examples
 
-### Outlined button example
-
-Source code API:
+API and source code:
 
 * `MaterialButton`
     * [Class description](https://developer.android.com/reference/com/google/android/material/button/MaterialButton)
     * [GitHub source](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/button/MaterialButton.java)
     
 The following example shows an outlined button with a text label and stroked container.    
 
-!["Outlined button example in Android with purple text surrounded by a gray outline"](assets/outlined-button.svg)
+!["Outlined button"](assets/outlined-button.svg)
 
+In the layout:
 ```xml
-    <Button
-        android:id="@+id/outlinedButton"
-        style="@style/Widget.MaterialComponents.Button.OutlinedButton"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Outlined button"
-        ...
-    />
+<Button
+    android:id="@+id/outlinedButton"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:text="Outlined button"
+    style="?attr/materialButtonOutlinedStyle"
+/>
+```
+
+_**Note:** `<Button>` is auto-inflated as `<com.google.android.material.button.MaterialButton>` via `MaterialComponentsViewInflater` when using a non-Bridge `Theme.MaterialComponents.*` theme._
```

Comment:
Is "inflated" a term used in the guidance? If not, can we define it?

Reply:
It's an Android-only developer term but I think this is quite important for them to know

Reply:
It's an Android-only developer term but I think this is quite important for them to know

Reply:
We probably should include a definition, though we should also agree on the baseline audience at some point.